libs/extensions/std/exec/src/gtfs-rt-interpreter-executor.ts

```diff
@@ -81,79 +122,173 @@ export class GtfsRTInterpreterExecutor
+                entity.tripUpdate.trip.routeId,
+              );
+              tripUpdate['entity.trip_update.stop_time_update.stop_sequence'] =
+                String(stopTimeUpdate.stopSequence);
+              tripUpdate['entity.trip_update.stop_time_update.stop_id'] =
+                String(stopTimeUpdate.stopId);
+              tripUpdate['entity.trip_update.stop_time_update.arrival.time'] =
+                String(stopTimeUpdate.arrival?.time);
+              tripUpdate['entity.trip_update.stop_time_update.departure.time'] =
+                String(stopTimeUpdate.departure?.time);
+
               rows.push(Object.values(tripUpdate) as string[]);
             }
           } else {
-            // TODO: Should we continue here or break?
-            resolve(
-              E.left(
-                new Error(
-                  'Error when parsing TripUpdates: Array stop_times_Update does not exist',
-                ),
-              ),
+            context.logger.logDebug(
+              `Parsing gtfs-rt feed data as TripUpdates: StopTimeUpdate of TripUpdate "${String(
+                entity.tripUpdate.trip.tripId,
+              )}" does not contain a single entry. Skipping this TripUpdate and continue with next one"`,
             );
+            continue;
           }
+
+          // Case: No TripUpdates found -> return sheet just with header
         } else {
-          // TODO: Should we continue here or break?
-          resolve(
-            E.left(
-              new Error(
-                'Error when parsing TripUpdates: TripUpdates does not exist',
-              ),
-            ),
+          context.logger.logDebug(
+            `Parsing gtfs-rt feed data as TripUpdates: No Tripupdates found in feedmessage with timestamp "${String(
+              feedMessage.header.timestamp,
+            )}"`,
           );
         }
       }
       resolve(E.right(new Sheet(rows)));
     });
   }
-}
 
-// This interface has to match with schema defined in downstream TableInterpreter
-interface TripUpdates {
-  'header.gtfs_realtime_version': string;
-  'header.timestamp': string;
-  'header.incrementality': string;
-  'entity.id': string;
-  'entity.trip_update.trip.trip_id': string;
-  'entity.trip_update.trip.route_id': string;
-  'entity.trip_update.stop_time_update.stop_sequence': string;
-  'entity.trip_update.stop_time_update.stop_id': string;
-  'entity.trip_update.stop_time_update.arrival.time': string;
-  'entity.trip_update.stop_time_update.departure.time': string;
+  private parseVehiclePositions(
+    feedMessage: GtfsRealtimeBindings.transit_realtime.FeedMessage,
+    context: ExecutionContext,
+  ): Promise<Either<Error, Sheet>> {
+    return new Promise((resolve) => {
+      context.logger.logDebug(
+        `Parsing raw gtfs-rt feed data as VehiclePosition"`,
+      );
+
+      const rows: string[][] = [];
+
+      // Add header
+      rows.push(Object.keys(new this.VehiclePosition()));
```

Comment:
I think in general this is the wrong way to model it. I did a bit of research and I think a record utility class might be nicer, see: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type

Consider the `type CatName` there as a place where you can put your hardcoded list of column names. Then you can model each row of the sheet as a record of that type and string (instead of CatName and CatInfo as is done in the example). I think that might be a better solution. Using classes seemed okay for me when we talked about this because I did not really know a better way but conceptually (and as done here) it is wrong.

Also, if you do not export a class it is already private to that file. So you could just put the classes in this file and not export them so they are only available here.

Reply:
This looks reeeally cool, got your point, thanks!

Then actually i would model it like this:
```
type TripUpdateHeader =
  | 'header.gtfs_realtime_version'
  | 'header.timestamp'
  | '...';
const row: Record<TripUpdateHeader, string> = {
  'header.gtfs_realtime_version': 'asdf',
  'header.timestamp': 'asdf',
  '...': 'asdf',
};
```

Reply:
Yeah exactly.